pkg/services/authn/clients/proxy.go

```diff
@@ -117,8 +107,34 @@ func (c *Proxy) Authenticate(ctx context.Context, r *authn.Request) (*authn.Iden
 	return nil, clientErr
 }
 
+// See if we have cached the user id, in that case we can fetch the signed-in user and skip sync.
+// Error here means that we could not find anything in cache, so we can proceed as usual
+func (c *Proxy) retrieveIDFromCache(ctx context.Context, cacheKey string, r *authn.Request) (*authn.Identity, error) {
+	entry, err := c.cache.Get(ctx, cacheKey)
+	if err != nil {
+		return nil, err
+	}
+
+	uid, err := strconv.ParseInt(string(entry), 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse user id from cache: %w - entry: %s", err, string(entry))
```

Comment:
There is a slight change of behavior there. Before this warn only.

Reply:
https://github.com/grafana/grafana/pull/83755/files#diff-6eadbf74eb9d29e650e8c49062cc08157568749dc108eca7f4a86190d761451eR93

logic for logging has been moved out of the function to reduce branching

Reply:
Ah, yeah i had misread that line.